### PR TITLE
feat: domains view in Wallets & Assets Page

### PR DESCRIPTION
**Pull request by 0xcen**
- Adds Domains category to Other Assets in Wallets & Assets view. 
- Fetches and adds domains to corresponding wallets in useTreasuryInfo hook as another asset class. 
 
Screenshot 2022-11-12 at 19 42 16

### CODE REVIEW

File: models/treasury/Domain.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export interface Domain {
+  owner: string
+  name: string | undefined
```
Comment (GregoryNEUT) [inline models/treasury/Domain.ts:3]: nit:

```suggestion
  name?: string
```

**Review by GregoryNEUT — COMMENTED** [review]
Hello! Have added few comments, hope it helps!

**Comment by 0xcen** [comment]
> Hello! Have added few comments, hope it helps!

Good feedback, thanks!

**Review by asktree — APPROVED** [review]
I didn't test anything, just did a style/best-practices review. I thought this code was clear and well written.

File: hooks/useTreasuryInfo/index.tsx [hunk 2]
```diff
@@ -33,8 +35,10 @@ export default function useTreasuryInfo(): Result<Data> {
   )
   const [nfts, setNfts] = useState<NFT[]>([])
   const [nftsLoading, setNftsLoading] = useState(true)
+  const [domainsLoading, setDoaminsLoading] = useState(true)
```
[the remark this answers is not in the document]
Comment (asktree) [reply hooks/useTreasuryInfo/index.tsx]: I don't think it's really needed to explicitly type `useState` when the inference is perfectly accurate already

File: components/treasuryV2/Details/DomainsDetails/Info/Domain.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+            )}
+          >
+            <ExternalLinkIcon className="h-3 w-3" />
+            Visit
+          </a>
+        </Link>
+        <DropdownMenu
+          triggerButton={
+            <button aria-label="Further Options">
+              <DotsVerticalIcon className=" h-4 w-4" />
```
[the remark this answers is not in the document]
Comment (asktree) [reply components/treasuryV2/Details/DomainsDetails/Info/Domain.tsx]: Haha I do this all the time. For me the vscode tailwind autocomplete often only works after the first character so I add a space to kick it in.

**Review by nramadas — APPROVED** [review]

**Comment by 0xcen** [comment]
> I transferred a domain to one of my multisig's wallets but it's not displaying in Realms. 
> 
> 
> 
> Multisig wallet address: GpHs7cegTDC7r1sy3zp2uUVRDarHQFJmMRcKEZC1YMZ7
> 
> 
> 
> UI on Realms: https://app.realms.today/dao/GgjTau55pSX5QRmGhfAGhQafHMAuDEwg4KRfaHuWXzr5/treasury/v2
> 
> 
> 
> Domain: https://naming.bonfida.org/domain/maddarken
> 
> 
> 
> Transfer on Solscan: https://solscan.io/tx/3ttF5exwN2mHwA3BkU85xvN9HPi4Jr5tTX3C7kJWat7ZQ3S9CVsasojC7SLtxJwQ6kKpTgs6TrkzgYxUFpxYZzzE

Hey Dan. This PR enables the domains view in your realms wallets. 

Until it's merged you can see your domain in the preview deployment. https://governance-p6gqwvihv-solana-labs.vercel.app

You can also transfer out the domain using the SNS Transfer Domain instruction when creating a proposal and your the domains in your wallets will also display there.

**Pull request merged**